File: 6th Semester/Programming Paradigms/NP1 Exercises - 4 programs 4 languages assignment/ex2.cs
using System;
  
public class ex2 {

    static public float leitorFloat(string message) {
        float numberFloat;
        while (true)
        {
            Console.Write(message);
            if (float.TryParse(Console.ReadLine(), out numberFloat)) return numberFloat;
            Console.WriteLine("Valor inválido. Tente novamente.");
        }
    }

    static public int leitorInt(string message) {
        int numberInt;
        while (true)
        {
            Console.Write(message);
            if (int.TryParse(Console.ReadLine(), out numberInt)) return numberInt;
            Console.WriteLine("Valor inválido. Tente novamente.");
        }
    }
  
    static public void Main()
    {

        Console.WriteLine("Exemplo: 86.4 195");
        float peso = leitorFloat("Insira seu peso em kg: ");
        float altura = (float) leitorInt("Insira sua altura cm: ") / 100;
        float imc = peso / (altura * altura);
        Console.Write(string.Format("Seu IMC calculado foi: {0:0.0}\nDe acordo com a OMS, sua classificação é: ", imc));
        if (imc < 18.5) Console.WriteLine("Abaixo do peso normal");
        else if (imc < 24.9) Console.WriteLine("Peso normal");
        else if (imc < 29.9) Console.WriteLine("Excesso de peso");
        else if (imc < 34.9) Console.WriteLine("Obesidade classe I");
        else if (imc < 39.9) Console.WriteLine("Obesidade classe II");
        else Console.WriteLine("Obesidade classe III");
    }
}

File: 6th Semester/Programming Paradigms/NP1 Exercises - 4 programs 4 languages assignment/ex2.py
# Desenvolva um programa para determinar se o indivíduo está com um peso favorável. Essa situação é
# determinada através do IMC (Índice de Massa Corpórea) onde o IMC=peso/altura², que é definida como sendo
# a relação entre o peso (PESO) e o quadrado da Altura (ALTURA) do indivíduo de acordo com a equação e tabela
# ao lado:

print('Exemplo: 86.4 195')
peso = float(input('Insira seu peso em kg: '))
altura = float(input('Insira sua altura em cm: ')) / 100
imc = peso / altura ** 2

print(f'Seu IMC calculado foi: {imc:.1f}\nDe acordo a OMS, sua classificação é: ', end="")
if imc < 18.5:
    print('Abaixo do peso normal')
elif imc < 24.9:
    print('Peso normal')
elif imc < 29.9:
    print('Excesso de peso')
elif imc < 34.9:
    print('Obesidade classe I')
elif imc < 39.9:
    print('Obesidade classe II')
else:
    print('Obesidade classe III')